baukit/nethook.py

```diff
@@ -105,7 +127,9 @@ def __exit__(self, type, value, traceback):
             return True
 
     def close(self):
-        self.registered_hook.remove()
+        if self.layer is not None:
+            # removing hook => restore the original forward function
+            self.module.forward = self.hook
```

Comment:
I think this isn't exactly the same as `del self.module.forward` - we should restore it cleanly in the common case where there was originally no instance-level `forward` reference.